File: Code/Cplus/LAB03_04/LAB03_04.cpp
#include <iostream>
#include <string>
using namespace std;

// Function to simulate the movement of the car based on sensor input
void controlCar(const string& sensorInput) {
    if (sensorInput == "left") {
        cout << "Turning left..." << endl;
    }
    else if (sensorInput == "right") {
        cout << "Turning right..." << endl;
    }
    else if (sensorInput == "center") {
        cout << "Moving straight..." << endl;
    }
    else {
        cout << "Invalid sensor input!" << endl;
    }
}

int main() {
    string sensorInput;

    // Example simulation
    while (true) {
        cout << "Enter sensor input (left, right, center): ";
        cin >> sensorInput;

        // Control the car based on sensor input
        controlCar(sensorInput);

        // Exit simulation
        if (sensorInput == "exit") {
            break;
        }
    }

    return 0;
}

File: Code/Python/LAB03_04.py
def control_car(sensor_input):
    if sensor_input == "left":
        print("Turning left...")
    elif sensor_input == "right":
        print("Turning right...")
    elif sensor_input == "center":
        print("Moving straight...")
    else:
        print("Invalid sensor input!")

def main():
    while True:
        sensor_input = input("Enter sensor input (left, right, center): ").strip().lower()
        
        # Control the car based on sensor input
        control_car(sensor_input)
        
        # Exit simulation
        if sensor_input == "exit":
            print("Exiting simulation...")
            break

if __name__ == "__main__":
    main()
